modules/health_quest/app/services/health_quest/patient_generated_data/options_builder.rb

```diff
@@ -34,30 +34,72 @@ def initialize(user, filters)
       # @return [Hash]
       #
       def to_hash
-        if appointment_id.present?
-          { subject: subject_reference }
-        else
-          { author: user.icn }
-        end
+        name = filters.delete(:resource_name)
+        query_param = filters.keys.pop
+
+        registry[name.to_sym][query_param.to_sym]
+      end
+
+      ##
+      # The registry which holds the return value for `to_hash`.
+      #
+      # @return [Hash]
+      #
+      def registry
+        {
+          questionnaire_response: {
+            appointment_id: { _tag: appointment_reference },
+            patient: { subject: user.icn },
+            authored: { authored: resource_created_date }
+          },
+          questionnaire: {
+            use_context: { 'context-type-value': context_type_value }
+          }
+        }
       end
 
       ##
       # Build the subject reference for the options that are being used to query the PGD for resources.
       #
       # @return [String]
       #
-      def subject_reference
-        "#{Settings.hqva_mobile.url}/appointments/v1/patients/#{user.icn}/Appointment/#{appointment_id}"
+      def appointment_reference
+        @appointment_reference ||=
+          "#{lighthouse.url}#{lighthouse.pgd_path}/NamingSystem/va-appointment-identifier|#{appointment_id}"
       end
 
       ##
-      # Get the appointment id from the filters that were passed into the controller action.
+      # Get the appointment id from the filters.
       #
       # @return [String]
       #
       def appointment_id
         @appointment_id ||= filters&.fetch(:appointment_id, nil)
       end
+
+      ##
+      # Get the authored date from the filters.
+      #
+      # @return [String]
+      #
+      def resource_created_date
+        @resource_created_date ||= filters&.fetch(:authored, nil)
+      end
+
+      ##
+      # Get the use context values from the filters.
+      #
+      # @return [String]
+      #
+      def context_type_value
+        '' # blank temp value
```

Comment:
This will be built out in the upcoming ticket: https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/19018